src/libponyc/ast/token.h

```diff
@@ -147,6 +147,7 @@ typedef enum token_id
   TK_IF,
   TK_IFDEF,
   TK_IFTYPE,
+  TK_IFTYPE_CLAUSE,
```

Comment:
I don't really like the name `TK_IFTYPE_CLAUSE` either, but I don't think `TK_IFTYPE_COND` is much better. This node contains both the condition and the clause.

I also had `TK_IFTYPE_BRANCH` in mind, which might be better. I'm open to other suggestions